docs(Minio): Add installation notes

nemesis1982 opened the pull request:
**Description**
- Adds installation notes
-- Adds a warning, PVC is stored on apps storage, this is a easily overlooked risk
-- Adds version information the guide is based on
-- Adds preparation steps
-- Adds minimal installation notes, including hostpath example
-- Adds information on domain requirement for S3
-- Adds ingress example, including API

⚒️ Fixes  # <!--(issue)-->

**⚙️ Type of change**

- [ ] ⚙️ Feature/App addition
- [ ] 🪛 Bugfix
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 🔃 Refactor of current code

**🧪 How Has This Been Tested?**
Documentation only, created based on discord thread: https://discord.com/channels/830763548678291466/1193205734729597030

Copied added content to website repository manually and ran yarn to verify.

**📃 Notes:**
<!-- Please enter any other relevant information here -->

**✔️ Checklist:**

- [ ] ⚖️ My code follows the style guidelines of this project
- [x] 👀 I have performed a self-review of my own code
- [ ] #️⃣ I have commented my code, particularly in hard-to-understand areas
- [x] 📄 I have made corresponding changes to the documentation
- [x] ⚠️ My changes generate no new warnings
- [ ] 🧪 I have added tests to this description that prove my fix is effective or that my feature works
- [ ] ⬆️ I increased versions for any altered app according to semantic versioning

**➕ App addition**

If this PR is an app addition please make sure you have done the following.

- [ ] 🪞 I have opened a PR on [truecharts/containers](https://github.com/truecharts/containers) adding the container to TrueCharts mirror repo.
- [ ] 🖼️ I have added an icon in the Chart's root directory called `icon.png`

---

_Please don't blindly check all the boxes. Read them and only check those that apply.
Those checkboxes are there for the reviewer to see what is this all about and
the status of this PR with a quick glance._

nemesis1982 committed:
Adds installation notes for minio

- Adds installation notes
-- Adds a warning, PVC is stored on apps storage, this is a easily overlooked risk
-- Adds version information the guide is based on
-- Adds preparation steps
-- Adds minimal installation notes, including hostpath example
-- Adds information on domain requirement for S3
-- Adds ingress example, including API

nemesis1982 committed:
Merge branch 'truecharts:master' into master

StevenMcElligott commented on charts/stable/minio/docs/installation_notes.md:
```diff
@@ -0,0 +1,37 @@
+  - `MINIO_ROOT_USER` and `MINIO_ROOT_PASSWORD` represent the username and password used to login to minio
+- Storage and Persistence->Integrated Persistent Storage->App Config Storage
+  - `Type of Storage` set to host path
+  - `Automatic Permissions Configuration`
```
Don't bother with Automatic Permissions, just have users set to `apps:apps` based on our guides

StevenMcElligott reviewed (changes requested):
I'd remove the automatic permissions, people should set proper permissions if using hostpath beforehand, if they have large datasets automatic permissions may take a long time

nemesis1982 committed:
Removed "Automatic Permissions Configuration" from minio installation page

- Removed Automatic permissions as this may be dangerous as pointed during review
- Added anther preparation step instead.

nemesis1982 commented:
> I'd remove the automatic permissions, people should set proper permissions if using hostpath beforehand, if they have large datasets automatic permissions may take a long time

Removed. Added a preparation step instead.

Crow-Control reviewed (changes requested):
ALL guides need to be clearly seperated by platform.

You didn't even mention "TrueNAS SCALE" once.

nemesis1982 committed:
Updates minio installation notes, adding platform note for TrueNAS SCALE

nemesis1982 commented:
> ALL guides need to be clearly seperated by platform.
> 
> You didn't even mention "TrueNAS SCALE" once.

Processed, this was not apparent from the examples I looked at nor the guidance I read. Can you send me a link to the requirements for the documentation as I obviously missed this.

StevenMcElligott commented:
I’ll fix / edit @nemesis1982, we do have guides that indicate what platform this guide is for, but we just have to add a bit about this guide applies to TrueNAS SCALE that’s all

StevenMcElligott commented on charts/stable/minio/docs/installation_notes.md:
```diff
@@ -0,0 +1,36 @@
+---
+title: Installation Notes
```
```suggestion
title: Installation Notes and Guide for TrueNAS SCALE
```

StevenMcElligott committed:
Update charts/stable/minio/docs/installation_notes.md

Signed-off-by: StevenMcElligott <89483932+StevenMcElligott@users.noreply.github.com>

StevenMcElligott commented on charts/stable/minio/docs/installation_notes.md:
```diff
@@ -0,0 +1,36 @@
+
+:::note Written for
+
+- Platform: TrueNAS SCALE
```
```suggestion
- Platform: TrueNAS SCALE Cobia 23.10.1
```

StevenMcElligott commented on charts/stable/minio/docs/installation_notes.md:
```diff
@@ -0,0 +1,36 @@
+  - `Type of Storage` set to host path
+  - `Host Path` select the dataset <minio_data>
+- Using Ingress and API ingress is recommended
+  - Main Ingress: minio.mydomain.com
```
```suggestion
  - Main Ingress (Also the Console) : minio.mydomain.com
```

StevenMcElligott committed:
Update charts/stable/minio/docs/installation_notes.md

Signed-off-by: StevenMcElligott <89483932+StevenMcElligott@users.noreply.github.com>

StevenMcElligott committed:
Update charts/stable/minio/docs/installation_notes.md

Signed-off-by: StevenMcElligott <89483932+StevenMcElligott@users.noreply.github.com>

stavros-k commented on charts/stable/minio/docs/installation_notes.md:
```diff
@@ -0,0 +1,36 @@
+- App Version: 2023.12.23
+- Chart Version: 9.1.0
+
+:::
```
```suggestion
---
title: Installation Notes
---

## TrueNAS SCALE

:::note Written for

- Platform: TrueNAS SCALE Cobia 23.10.1
- App Version: 2023.12.23
- Chart Version: 9.1.0

:::
```

StevenMcElligott replied:
Sounds good!

stavros-k reviewed (commented):
@StevenMcElligott 

I think this is  a bit cleaner, we can add above 

```md
## Helm

... content ...
```

or other platforms

stavros-k committed:
Update charts/stable/minio/docs/installation_notes.md

Signed-off-by: Stavros Kois <47820033+stavros-k@users.noreply.github.com>

stavros-k commented:
@all-contributors please add @nemesis1982  for docs
